server/routerlicious/packages/services-client/src/array.ts

```diff
@@ -30,3 +32,72 @@ export function convertSortedNumberArrayToRanges(numberArray: number[]): number[
 	ranges.push([begin, end]);
 	return ranges;
 }
+
+export function mergeArrays<T, TComp>(arr1: T[], arr2: T[], selector: (item: T) => TComp): T[] {
```

Comment:
Not sure if we can use the built-in method? https://blog.mikaeels.com/how-to-merge-arrays-in-typescript

Maybe name this method as mergeArraysAscend?

Reply:
I think calling this `mergeSortArrays` would be better since that's the algorithm being used, and `selector` should be [`compareFn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#comparefn).

However, I also agree with Joel that built-ins would be simpler and safer, even though they might be _slightly_ less efficient.

```ts
export function mergeAndSortArrays(): <T>(arr1: T[], arr2: T[], compareFn: (item: T) => number): T[] {
  return [...arr1, ...arr2].sort(compareFn);
}
```

Reply:
I agree that `mergeArraysAscending` would be a more accurate name. I think doing manual mergeSort with the selector makes sense.

Using built-ins would look like this, but IDK if it's worth the performance penalty of using more memory and iterating twice, unless JS does something fancy in the background to optimize merge and sort one-liners.
```ts
export function mergeArraysAscending<T, TComp>(arr1: T[], arr2: T[], selector: (item: T) => TComp): T[] {
  const compareFn = (a: T, b: T) => {
    return selector(a) - selector(b);
  };
  return [...arr1, ...arr2].sort(compareFn);
}
```

Obviously cleaner, but also less efficient.

Reply:
Yeah, build something basic data structure for best performance. Changed the `selector` to use `comparator` instead